sound/soc/sof/pm.c

```diff
@@ -178,6 +178,18 @@ static int sof_resume(struct device *dev, bool runtime_resume)
 			dev_err(sdev->dev, "ctx_restore IPC error during resume: %d\n", ret);
 	}
 
+setup_fail:
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_ENABLE_DEBUGFS_CACHE)
+	if (ret < 0) {
+		/*
+		 * Debugfs cannot be read in runtime suspend, so cache
```

Comment:
I read that comment multiple times, and I don't think it's correct.

We *can* read debugs, it's just that in pm_runtime suspend the cache contents will be provided.

This comment confuses cause and effect IMHO.

the commit message has the same issue, it's not "impossible to retrieve an exception", it's rather "the exception is not available in the cached data".

I may be splitting hair on this one but the logic seems off.

Reply:
@plbossart the issue is I think we cannot force DSP to remain in D0 if it fails to enter D0, which means we cannot collect core dumps because the DSP is left in D3. This caches them on failure so that we can still get them even though DSP never entered D0.

Reply:
I get the issue, what I argue on is the comments and explanations. We can *always* read debugfs, it's only the cached results that need to be forced.

Reply:
Regarding your comment though, in the linked failure state, it is impossible to fetch the failure state, with or without cache without this change.

With cache you get nothing
Without cache you get told DSP is in wrong state and get nothing (i.e cannot be read)

Reply:
Thanks, let me rephrase the commit message and reupload.